### PR TITLE
Add gateio sub accounts

### PR DESCRIPTION
**Pull request by sc0Vu**
Add gateio sub accounts:
https://www.gate.io/docs/developers/apiv4/en/#subaccount

In this PR, I also update `defineRestApiEndpoint`, support empty string and xxx_xxx key in apis object.
eg: `privateSubAccountsGet`, `private_sub_accounts_get`

```
'sub_accounts': {
  '': 1,
}
```

### CODE REVIEW

File: js/gate.js [hunk 1]
```diff
@@ -210,6 +212,26 @@ module.exports = class gate extends Exchange {
                             'sub_account_transfers': 300,
                         },
                     },
+                    'sub_accounts': {
```
Comment (carlosmiei) [inline js/gate.js:215]: @sc0Vu why not do something like this?

```Javascript

(...)
'subAccounts': 'https://api.gateio.ws/api/v4'
(,,,)

'subAccounts': {
     'get': {
            'sub_accounts':1,
            'sub_accounts/{user_id}:1,
             (....)
     }
}

```
I think that by doing this ⬆️ you don't have to change anything, and we avoid a weird format, lmk what you think

Comment (sc0Vu) [reply js/gate.js:215]: @carlosmiei you're right, I also consider about this:

```
(...)
'subAccounts': 'https://api.gateio.ws/api/v4/sub_accounts'
(...)

'subAccounts': {
     'get': {
            '':1,
            '{user_id}':1,
             (....)
     }
}
```

In this way, user don't need to type another SubAccounts.

Comment (carlosmiei) [reply js/gate.js:215]: I understand your point, but I'd rather have to duplicate the path a bit than have an empty key, what do you think? @kroitor

Comment (sc0Vu) [reply js/gate.js]: @carlosmiei  I removed the updates in Exchange.js.

File: js/gate.js [hunk 2]
```diff
@@ -2823,7 +2845,7 @@ module.exports = class gate extends Exchange {
             request['chain'] = network;
             params = this.omit (params, 'network');
         }
-        const response = await this.privateWithdrawalsPost (this.extend (request, params));
+        const response = await this.privateWithdrawalsWithdrawalsPost (this.extend (request, params));
```
Comment (carlosmiei) [inline js/gate.js:2848]: @sc0Vu I think you have a typo here, should be `privateWithdrawalsPostWithdrawals` right?

Comment (sc0Vu) [reply js/gate.js]: You're right, fixed!

**Comment by carlosmiei** [comment]
LGTM ✅

**Review by carlosmiei — APPROVED** [review]

**Comment by ndubel** [comment]
It don't works for me:
```
fetch Request:
 gateio GET https://api.gateio.ws/api/v4/subAccounts/sub_accounts
RequestHeaders:
 {
  'X-Gate-Channel-Id': 'ccxt',
  KEY: 'xxx',
  Timestamp: '1666628405',
  SIGN: 'xxx',
  'Content-Type': 'application/json'
}
RequestBody:
 undefined

PL Subscribed for 1 redis channels
handleRestResponse:
 gateio GET https://api.gateio.ws/api/v4/subAccounts/sub_accounts 404 Not Found
ResponseHeaders:
 {
  Connection: 'keep-alive',
  'Content-Encoding': 'gzip',
  'Content-Type': 'text/html',
  Date: 'Mon, 24 Oct 2022 16:20:06 GMT',
  Server: 'openresty',
  'Transfer-Encoding': 'chunked'
}
ResponseBody:
 <html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>openresty</center>
</body>
</html>


ExchangeNotAvailable: gateio GET https://api.gateio.ws/api/v4/subAccounts/sub_accounts 404 Not Found <html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>openresty</center>
</body>
</html>
```
request url is https://api.gateio.ws/api/v4/subAccounts/sub_accounts not https://api.gateio.ws/api/v4/sub_accounts

**Comment by ndubel** [comment]
@sc0Vu @carlosmiei @kroitor can you check the issue above?

**Comment by carlosmiei** [comment]
@ndubel sure, we will take a look sorry for the delay

**Comment by carlosmiei** [comment]
@ndubel it should be working now, can you please upgrade to the latest version and try again?

**Comment by ndubel** [comment]
@carlosmiei works great, thank you

**Pull request merged**